contrib/run_task.sh

```diff
@@ -146,20 +198,32 @@ loop_features() {
     fi
 }
 
-# Lint the workspace then the individual crate examples.
+# Lint the workspace, lint the individual crate examples, check for duplicate dependencies.
 do_lint() {
     need_nightly
 
-    # Use the current (recent/minimal) lock file.
     local cargo="cargo --locked"
 
-    $cargo clippy --workspace -- -D warnings
+    $cargo clippy --workspace --all-features --all-targets -- -D warnings
```

Comment:
In d457ed8d54d72fdb7e09b69d38527f864f022a75:

Let's add a couple more clippy lines (at least, one without `--all-features`, as well as one with `--no-default-features` if that doesn't trigger any errors related to nostd)

Reply:
Turns out we don't need to lint the examples separately, I guess I got confused at some stage in the past. I've now used:
```rust
    # Lint various feature combinations to try and catch mistakes in feature gating.
    $cargo clippy --workspace --all-targets --keep-going -- -D warnings
    $cargo clippy --workspace --all-targets --all-features --keep-going -- -D warnings
    $cargo clippy --workspace --all-targets --no-default-features --keep-going -- -D warnings
```

Note also, TIL and used `--keep-going`